extensions/wikia/Flags/controllers/FlagsController.class.php

```diff
@@ -80,24 +80,60 @@ public function getFlagsForPageForEdit() {
 		if ( !isset( $this->params['pageId'] ) ) {
 			return null;
 		}
+		$wikiId = $this->params['wikiId'];
+		$pageId = $this->params['pageId'];
+
+		$flagsCache = new FlagsCache();
+		$allFlagTypes = $flagsCache->getFlagsForPageForEdit( $wikiId, $pageId );
+
+		if ( !$allFlagTypes ) {
+			/**
+			 * 1. Get all flag types with instances for the page
+			 */
+			$allFlagTypes = $this->getAllFlagTypes( $wikiId, $pageId );
+
+			/**
+			 * 2. Create links to templates for creation of "See more" links
+			 */
+			foreach ( $allFlagTypes as $flagTypeId => $flagType ) {
+				$title = Title::newFromText( $flagType['flag_view'], NS_TEMPLATE );
+				$allFlagTypes[$flagTypeId]['flag_view_url'] = $title->getFullURL();
+			}
+
+			/**
+			 * 3. Cache the results and set the response data
+			 */
+			$flagsCache->setFlagsForPageForEdit( $wikiId, $pageId, $allFlagTypes );
+		}
+
+		$this->setResponseData( $allFlagTypes );
+	}
+
+	private function getAllFlagTypes( $wikiId, $pageId ) {
+		$flagsCache = new FlagsCache();
```

Comment:
What do you think about making global variable with FlagCache and method to get it? We are creating two instance of FlagCache in one algorithm iteration